indexd/index/drivers/query/urls.py

```diff
@@ -44,6 +44,8 @@ def query_urls(
             versioned.lower() in ["true", "t", "yes", "y"] if versioned else None
         )
 
+        any_authz, authz = _enforce_rbac([])
```

Comment:
we need a better name for all these (the two variables and the function) - I can't tell what their intention is.

is this something like:

`user_authz_resources, ?? what is authz`? and why are we passing an empty list?

Let's logically separate these maybe and refactor into something to get the authz resources if you need them (which it looks like you do).

Reply:
and let's not do any of this unless we need to (e.g. record discovery is enabled) and let's make that clear at the high-level here. 

I would move all this down to the section where you have `if any_authz` and instead the if condition should be `if ARE_RECORDS_DISCOVERABLE is False` and then do all the new logic

Reply:
actually.. the driver should not be relying on a stateful request at all. We need to refactor the portion where we obtain user authz out of this and pass it in as `filter_authz_resources` in here. That'd actually be much cleaner - no config checks at all in here, just add the filter if you need to, calling code determines if there needs to be a filter at all.

Reply:
> actually.. the driver should not be relying on a stateful request at all.  

I had a version that did this - checked config in the calling code.  Unfortunately, as far as I can see, that means adding the code to all the blueprints both to check config and change calls.

Reply:
other points except dependency on stateful request - done